packages/format/src/filter.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { Result } from 'axe-core';
+
+/**
+ * Filter to post-process a11y results from axe
+ */
+export interface Filter {
+    (violations: Result[], ...args: never[]): Result[];
+}
+
+// TODO (refactor): constrain rule id to known rule ids e.g using string literal, keyof, in etc
+//  e.g. https://stackoverflow.com/a/54061487
+// const ruleIDs = getRules().map((ruleObj) => ruleObj.ruleId);
+// type ruleID = keyof ruleIDs;
+// type ruleID = typeof ruleIDs[number];
+type ruleID = string;
+type cssSelectors = string[];
+
+/**
+ * Exception list of map of rule to corresponding css targets that needs to be filtered from a11y results.
+ */
+export type exceptionList = Record<ruleID, cssSelectors>;
+
+/**
+ * Filter a11y violations from axe based on given {@link exceptionList}
+ * @param violations - List of violations found with axe
+ * @param exceptionList - {@link exceptionList} of map of rule to corresponding css targets that needs to be filtered from a11y results
+ */
+export function exceptionListFilter(violations: readonly Result[], exceptionList: exceptionList = {}): Result[] {
+    const exceptionRules = Object.keys(exceptionList);
+    if (exceptionRules.length === 0) return violations as Result[];
+
+    const filteredViolations: Result[] = [];
+
+    for (const violation of violations) {
+        if (!exceptionRules.includes(violation.id)) {
+            filteredViolations.push(violation);
+        } else {
+            for (const result of violation.nodes) {
+                const filteredResults = result.target.filter(
+                    (cssSelector) => !exceptionList[violation.id].includes(cssSelector)
```

Comment:
I'm unclear how exactly this CSS selector works. How exactly is this selector determined? Since the caller must pass in a matching value, how do they know what axe produces here?  

Just thinking through the full flow here and how often the value will need to be updated as the DOM of the page changes over time.

Reply:
Got it. My concern was really around how axe generates the selectors and how often they change. If the selector axe returns here changes every time the DOM structure changes, then I think this will be a maintenance nightmare to keep up with. If the selectors are pretty generic and rarely change then it shouldn't be a problem.

Reply:
Agree, it is a valid concern. But I don't think there are any workarounds as CSS selectors are what we have as handles to UI components. When there are multiple selectors guess we can choose one which is less dependent on the DOM structure.